### ISSUE TITLE
Suffix match order bug

### ISSUE THREAD

**Issue by ChristopherChudzicki** [mitodl/mitx-grading-library#132]
Note: I came across this bug while rewriting the parser/evaluator. I'm creating the issue mostly so I can reference it in a future PR. (A PR which would also fix this bug.)

```python
from mitxgraders.helpers.calc import evaluator

variables = {}
functions = {}
suffixes_ok = {  'ab': 2 }
suffixes_error = { 'a': 10, 'ab': 2 }

expr = '5ab'

evaluator(expr, variables, functions, suffixes_ok) # works as expected
evaluator(expr, variables, functions, suffixes_error) # throws an error
```

The issue is that `a` is a front substring of `ab`. I think that the grammar was originally intended to handle this case because in defining suffixes, `MatchFirst` is used, which can be used to specify the order in which to check. (So above, `MatchFirst(['ab', 'a'])` would have been fine.) But we give `MatchFirst` a bunch of dictionary keys, which aren't ordered:

```python
# Define our suffixes
suffix = MatchFirst(Literal(k) for k in self.suffixes.keys())
```

**Comment by ChristopherChudzicki** [comment]
Incidentally, it is also possible to do some pretty silly things with suffixes:

```python
import numpy as np
from mitxgraders.helpers.calc import evaluator

variables = {}
functions = {}
suffixes_ok = { '-1': 10 }

expr = '5-1'

result, _ = evaluator(expr, variables, functions, suffixes_ok)
print(result) # 10.0
```

**Comment by jolyonb** [comment]
You sure that doesn't give you 50?

I was aware the suffixes had these issues, but since we were restricting them to % and hard-coded metric suffixes, it seemed safe enough. Feel free to fix if you like. I suspect that it should probably restrict to single character suffixes that cannot include certain characters.

**Comment by jolyonb** [comment]
Resolved by #134
